torch/_functorch/aot_autograd.py

```diff
@@ -19,6 +19,9 @@
 from torch import Tensor
 from torch._dispatch.python import enable_python_dispatcher
 from torch._dynamo.utils import dynamo_timed
+from torch._dynamo.logging import AOTForwardGraphLogRec, AOTJointGraphLogRec, AOTBackwardGraphLogRec
```

Comment:
I didn't review this part of the design too carefully when it was originally put out. Seeing it now, it seems a little boilerplatey that you have to pre-declare all of the artifacts in order to log them. Not sure if I have a concrete suggestion otherwise though.

Reply:
Yeah the "dependency" here is that you need to know the name that's used in the env var and API for each type => some sort aot artifact registration.